kotlinx-coroutines-core/common/test/flow/operators/BufferConflationTest.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.flow
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import kotlin.test.*
+
+/**
+ * A _behavioral_ test for conflation options that can be configured by [buffer] operator to test that it is
+ * implemented properly and that adjacent [buffer] calls are fused properly.
+*/
+class BufferConflationTest : TestBase() {
+    private val n = 100 // number of elements to emit for test
+
+    private fun checkConflate(
+        capacity: Int,
+        onBufferOverflow: BufferOverflow = BufferOverflow.DROP_OLDEST,
+        op: suspend Flow<Int>.() -> Flow<Int>
+    ) = runTest {
+        expect(1)
+        // emit all and conflate / then collect first & last
```

Comment:
```suggestion
        // emit all and conflate, then collect first & last
```
Also, this comment might be outdated.